scripts/quests/sandoria/PLD_AF2_A_Boys_Dream.lua

```diff
@@ -0,0 +1,235 @@
+                    quest:setVar(player, 'Prog', 4)
+                end,
+
+                [25] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 7)
+                end,
+
+                [47] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 5)
+                end,
+            },
+
+        },
+
+        [xi.zone.SOUTHERN_SAN_DORIA] =
+        {
+            ['Exoroche'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if
+                        questProgress == 2 and
+                        player:getMainJob() == xi.job.PLD
+                    then
+                        return quest:progressEvent(50)
+                    elseif questProgress >= 7 then
+                        return quest:progressEvent(32)
+                    else
+                        player:startEvent(76)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [32] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 8)
+                end,
+
+                [50] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 3)
+                end,
+            },
+        },
+
+        [xi.zone.SELBINA] =
+        {
+            ['Zaldon'] =
+            {
+                onTrade = function(player, npc, trade)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if
+                        questProgress >= 4 and
+                        npcUtil.tradeHasExactly(trade, xi.items.ODONTOTYRANNUS)
+                    then
+                        return quest:progressEvent(85)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [85] = function(player, csid, option, npc)
+                    npcUtil.giveKeyItem(player, xi.ki.KNIGHTS_BOOTS)
+                    quest:setVar(player, 'Prog', 6)
+                    player:confirmTrade()
+                end,
+            },
+        },
+
+        [xi.zone.CHATEAU_DORAGUILLE] =
+        {
+            ['_6h0'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 8 then
+                        return quest:progressEvent(88)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [88] = function(player, csid, option, npc)
+                    if
+                        player:getMainJob() == xi.job.PLD and
+                        quest:complete(player)
+                    then
+                        player:delKeyItem(xi.ki.KNIGHTS_BOOTS)
+                        return quest:event(90)
```

Comment:
Does this work? We've already returned the quest object when calling event 88

Reply:
yes it works. i spent a ton of time on this part